### DISCUSSION TITLE
Version 1.4

### DISCUSSION THREAD

**Discussion by ahadas** [mucommander/mucommander#1136] [General]
as I should have more time for muCommander over the next two weeks, I'd like to release version 1.4.0.
looking at [the remaining items in the backlog](https://github.com/mucommander/mucommander/milestone/15):

#954 - we didn't get confirmation from the reporter, but from the feedback on the issue and tests I ran locally, the fix seems to work so we will only need to publish the key along with the RPM

#492 - @pskowronek @Andreas0602, how comfortable are you with the remaining changes in this PR? we merged quite a few changes in separate PRs so specifically about those that we not merged yet, did you use a version with these changes long enough to say that it works reasonably way?

#962 - I've made the preparation for adding another type of JRE for macOS/aarch64. We got enough donations to acquire an account for signing the application for macOS  (for two years, thanks everyone who made a donation!) - so my plan is to see if we can build muCommander + build [our customized JDK](https://github.com/mucommander/JDK) + sign the app with GitHub actions to address this issue

and besides those items, any concern or something I missed?

**Comment by ghost** [comment]
> https://github.com/mucommander/mucommander/issues/492 - @pskowronek @Andreas0602, how comfortable are you with the remaining changes in this PR? we merged quite a few changes in separate PRs so specifically about those that we not merged yet, did you use a version with these changes long enough to say that it works reasonably way?

My Mac with an Intel processor broke down some time ago, and I‘m not able to build muCommander on my new Mac with Apple silicon. Therefore I cannot comment on this PR.

By the way, it would be great if muCommander would support Apple silicon soon. There aren’t many apps left for the Mac that don't natively support Apple silicon nowadays. muCommander should keep up with the times.

**Reply by ahadas** [reply]
> > #492 - @pskowronek @Andreas0602, how comfortable are you with the remaining changes in this PR? we merged quite a few changes in separate PRs so specifically about those that we not merged yet, did you use a version with these changes long enough to say that it works reasonably way?
> 
> My Mac with an Intel processor broke down some time ago, and I‘m not able to build muCommander on my new Mac with Apple silicon. Therefore I cannot comment on this PR.

ack
I ran muCommander on aarch64 for sure but I don't remember if I also tried to build muCommander there.. do you remember what was the issue?

> By the way, it would be great if muCommander would support Apple silicon soon. There aren’t many apps left for the Mac that don't natively support Apple silicon nowadays. muCommander should keep up with the times.

yep, that's the idea in what I wrote about #962  :wink:

**Reply by ghost** [reply]
> do you remember what was the issue?

The app terminated itself within less than one second after starting. I guess that was because the dmg file contained the Java runtime environment for x86_64, whereas the Java compiler was for aarch64.

**Reply by pskowronek** [reply]
@ahadas  - I know that muC creates DMG using some custom fork of macAppBundle, but FYI - recently I've provided PR to macAppBundle, author doesn't maintain it anymore, however he accepted my PR: https://github.com/crotwell/gradle-macappbundle/pull/75 and released a new version: 2.3.1. I've removed native runners and all now fall back to use bash script to find JRE and execute jar with it + fixes to bash script to work on Ventura+. Just to let you know - it might help with running muC on mac silicon, but JRE must be installed in the system.

**Reply by ahadas** [reply]
> The app terminated itself within less than one second after starting. I guess that was because the dmg file contained the Java runtime environment for x86_64, whereas the Java compiler was for aarch64.

something may have changed then because I tested muCommander on @danielerez's machine with M1 and the app started successfully with JRE for x64, but it's been long ago... ok, let's create a build of 1.4.0 with JRE for aarch64 and see

**Reply by ahadas** [reply]
@pskowronek please note that [gradle-macappbundle was dropped from muCommander](https://github.com/mucommander/mucommander/commit/7fa7b41ab98f8844f0703a3b3385dcf032f7b09d) in favor of jpackage

**Reply by ahadas** [reply]
> ... because I tested muCommander on @danielerez's machine with M1 and the app started successfully with JRE for x64, but it's been long ago

actually, I was wrong about this once again - @danielerez's machine doesn't have M1 but x64, it just had a newer version of macOS than my machine had back then

**Reply by ahadas** [reply]
ok, so the nightly builds include a dmg for aarch64 and @Andreas0602 confirmed it works fine on his machine so this part is covered

**Comment by pskowronek** [comment]
> #492 - @pskowronek @Andreas0602, how comfortable are you with the remaining changes in this PR? we merged quite a few changes in separate PRs so specifically about those that we not merged yet, did you use a version with these changes long enough to say that it works reasonably way?

I would be comfortable if it was checked on different OS types and machines. I don't run muC with that PR on daily basis, but when I do it runs no problem - however, yesterday I've spotted one race condition related to tooltips for terminal slider (it doesn't break the main functionality tho) - I will try to provide the fix today. 

So, the question is - can we provide beta build with that PR (or nightly to include that)? And from your experience @ahadas  - how many people run such builds?

**Reply by ghost** [reply]
I'm using muCommander also at work (alongside Double Commander) on a Windows 10 laptop. I could test a build with the PR there. But I'd need an MSI installer as I can't manually build muCommander on my work machine.

In my view, the PR should be merged into the master branch as soon as possible, and then the nightly build can be easily tested by anyone interested. Any issues identified could then be fixed directly on the master branch. A faster startup has a high priority. On my Windows laptop the startup is very slow -- much slower than on my Mac. Not including the PR in v1.4 is not a good option. Moving forward with integration seems best. And anyone using nightly beta builds should be aware anyway that they may be unstable and contain bugs.

**Reply by pskowronek** [reply]
@ahadas are you able to create MSI for @Andreas0602?

**Reply by ahadas** [reply]
@pskowronek it's hard to say how many people run unstable builds - some were downloaded dozens of times but most of them were not downloaded at all for a long time

I agree with @Andreas0602 that producing the nightly build with these changes is the best way forward but I do want to take another look at the changes, I'll try to do it this week. if @Andreas0602 and we test that build and it's good enough then I think it's enough

**Reply by ahadas** [reply]
#492 was merged and included in the nightly builds

**Comment by ahadas** [comment]
alrighty, seems we're good to go with 1.4.0. I'm going to make few changes in the readme file and such, and intend to release it this week

**Comment by pskowronek** [comment]
@ahadas just installed a released 1.4.0 on mac - so far so good, but when I start it, a message appears that a newer version is available :) just letting you know, maybe you're in the middle of release and that's why it is happening.

**Reply by ahadas** [reply]
Yeah, thanks, I'm on it

**Reply by ahadas** [reply]
should be fixed now

**Reply by pskowronek** [reply]
it works now

**Comment by pskowronek** [comment]
First bug found in 1.4.0 - rather cosmetic one: #1162

**Reply by pskowronek** [reply]
hm, might be related to my OCLP Ventura - @ahadas are you able to check if you reproduce it?

**Comment by PeterPlanlos** [comment]
Hello there, maybe it is the wrong place but what is -1 in your release versions on your github repo?
example: https://github.com/mucommander/mucommander/releases/download/1.4.0-1/mucommander-1.4.0-1-x86_64.dmg

the CFBundleShortVersionString or CFBundleVersion from the installed app on macOS shows 1.4.0, you can get this information from 
/Applications/muCommander.app/Contents/Info.plist. makes very hard to compare the installed version with the newest from your github repo.
Does the -1 on the releaes ever change to -2 again? why not just update to 1.4.1 if you fix anything or push a small update?

thank you very much and keep up the great work. Thanks 👍

**Reply by ahadas** [reply]
@PeterPlanlos that's due to the convention of versions in linux which includes a build-number, i.e., the first build of 1.4.0
we rarely create a second build for a version, if I recall correctly, we did it only once when one of the packages needed to be rebuilt
the next version is supposed to be 1.5.0. if we need to fix something on top of 1.4.0 then we'll have 1.4.1. and if one of the packages need to be rebuilt, we'll have 1.4.0-2 for that package
